src/scalar.rs

```diff
@@ -270,12 +270,33 @@ impl Scalar {
     /// require specific bit-patterns when performing scalar
     /// multiplication.
     pub const fn from_bits(bytes: [u8; 32]) -> Scalar {
-        let mut s = Scalar { bytes };
-        // Ensure that s < 2^255 by masking the high bit
-        s.bytes[31] &= 0b0111_1111;
+        let mut bytes:[u8; 32] = bytes;
+        bytes[31] &= 0b0111_1111;
+        Scalar { bytes }
+    }
 
-        s
+    /// Construct a `Scalar` from the low 255 bits of a little-endian 256-bit integer
+    /// `clamping` it's value to be in range
+    ///
+    /// **n ∈ 2^254 + 8\*{0, 1, 2, 3, . . ., 2^251 − 1}**
+    pub const fn from_bits_clamped(bytes: [u8; 32]) -> Scalar {
+        Scalar{ bytes : Scalar::clamp(bytes) }
+    }
+
+    /// Constructs `[u8; 32]` bytes representing 255 bit little-endinan integer in range
+    ///
+    /// **n ∈ 2^254 + 8\*{0, 1, 2, 3, . . ., 2^251 − 1}**
+    ///
+    /// input value `bytes` is treated as little-endian 256-bit integer
+    pub const fn clamp(bytes: [u8; 32]) -> [u8; 32] {
+        let mut bytes : [u8; 32] = bytes;
+        bytes[0]  &= 0b1111_1000;
+        bytes[31] &= 0b0111_1111;
+        bytes[31] |= 0b0100_0000;
+
+        bytes
     }
```

Comment:
Several things here:

1. This is an inherent method of `Scalar` but it doesn't actually deal with the `Scalar` type? That's weird
2. It's once again making an unnecessary copy of `bytes`
3. I don't think there's a use case for clamping the bytes without using them to construct a `Scalar`?

I'd suggest just getting rid of it and folding it into a function that clamps and constructs a `Scalar`.

Reply:
removed.